src/main/java/uk/gov/hmcts/divorce/payment/PaymentService.java

```diff
@@ -326,4 +330,12 @@ public Double getServiceCost(String service, String event, String keyword) {
 
         return feeResponse.getAmount();
     }
+
+    private boolean isGenericErrorRequiredForHttpStatus(HttpStatus httpStatus) {
+        boolean required = switch (httpStatus) {
+            case INTERNAL_SERVER_ERROR, GATEWAY_TIMEOUT -> true;
+            default -> false;
+        };
+        return required;
```

Comment:
How about using ||? There are only 2 conditions:
```suggestion
        return httpStatus == INTERNAL_SERVER_ERROR || httpStatus == GATEWAY_TIMEOUT;
```

Reply:
Had started with this but was wondering if in future we will need more custom error messages based on HttpStatus. If that could be the case, I could have this function return the error message itself based on value of httpStatus?
The ticket however mentions only 2 codes (500, 504)